### PR TITLE
sql/postgres: ent based revisions for postgres

### PR DESCRIPTION
**Pull request by masseelch**

### CODE REVIEW

File: sql/internal/sqlx/migrate.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+			SetExecutedAt(rev.ExecutedAt).
+			SetExecutionTime(rev.ExecutionTime).
+			SetHash(rev.Hash).
+			SetOperatorVersion(rev.OperatorVersion).
+			SetMeta(rev.Meta)
+	}
+	return r.c.Revision.CreateBulk(bulk...).
+		OnConflict(
+			sql.ConflictColumns(revision.FieldID),
+		).
```
Comment (a8m) [inline sql/internal/sqlx/migrate.go:79]: Why do you expect conflict? In case of retrying?

File: sql/internal/sqlx/migrate.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	c *ent.Client
+	// sc is the function signature of the Ent migration engine.
+	// Due to cyclic dependencies between Ent and Atlas, this value is stitched in at runtime.
+	sc func(context.Context) error
+}
+
+// NewRevisionStorage creates a new RevisionStorage with the given ent.Client.
+func NewRevisionStorage(c *ent.Client) *RevisionStorage {
+	return &RevisionStorage{c: c}
+}
```
Comment (a8m) [inline sql/internal/sqlx/migrate.go:27]:
```suggestion
func NewRevisionStorage(dialect string, db schema.ExecQuerier) *RevisionStorage {
	return &RevisionStorage{
		c: ent.NewClient(ent.Driver(sql.NewDriver(sql.Conn{ExecQuerier: db}, dialect))),
	}
}
```

Then, in drivers, call:

```go
sqlx.NewRevisionStorage(dialect.Postgres, db)
sqlx.NewRevisionStorage(dialect.MySQL, db)
```

File: sql/internal/sqlx/migrate.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	"context"
+
+	"ariga.io/atlas/sql/migrate"
+	"ariga.io/atlas/sql/migrate/ent"
+	"ariga.io/atlas/sql/migrate/ent/revision"
+	"entgo.io/ent/dialect/sql"
+)
+
+// A RevisionStorage provides implementation for the migrate.RevisionReadWriter interface.
+type RevisionStorage struct {
```
Comment (a8m) [inline sql/internal/sqlx/migrate.go:17]:
```suggestion
type EntRevision struct {
```

sounds better IMO for an ent-based implementation.

File: sql/internal/sqlx/migrate.go (hunk 1 above)
Comment (masseelch) [reply sql/internal/sqlx/migrate.go:79]: I always do an insert and when there already is data I replace it.

Comment (a8m) [reply sql/internal/sqlx/migrate.go:80]: Something I don't understand about the product flow. In which case one execution overrides the previous, on retrying? Don't you think it makes sense to store the previous execution information as well? I'd expect to see this in the UI as well.

@rotemtam, thoughts?

**Review by a8m — APPROVED** [review]

**Pull request merged**